core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/cassandra/MetricsServiceCassandra.java

```diff
@@ -800,6 +801,40 @@ private void dropKeyspace(String keyspace) {
         });
     }
 
+    @Override
```

Comment:
That's a good point. For now, what about updating SimpleDataCallback to return a 204 for either null or an empty collection. I say **for now** because I don't like returning nulls, and it has to do with how some of the queries and mapper classes are implemented. NumericMetricMapper for example returns null when there is no data. It cannot return the NumericMetric object having no data because it doesn't have the tenant id or the metric id. But we already know those. I think this should be addressed separately though.